src/gameLogic/generation/actions/ActionGenerator_NPC.cpp

```diff
@@ -20,19 +20,26 @@ namespace spy::gameplay {
     std::shared_ptr<BaseOperation>
     ActionGenerator::generateRandomAction(const State &s, const util::UUID &activeCharacter,
                                           const MatchConfig &config) {
-        auto move = generateMovementActions(s, activeCharacter);
-        auto properties = generateAllPropertyActions(s, activeCharacter);
-        auto gadget = generateAllGadgetActions(s, activeCharacter, config);
-        auto spy = generateSpyActions(s, activeCharacter);
-        auto gamble = generateGambleActions(s, activeCharacter);
-
-        auto ret = move;
-        ret.insert(ret.end(), properties.begin(), properties.end());
-        ret.insert(ret.end(), gadget.begin(), gadget.end());
-        ret.insert(ret.end(), spy.begin(), spy.end());
-        ret.insert(ret.end(), gamble.begin(), gamble.end());
-        ret.insert(ret.end(), generateRetire(activeCharacter));
-
-        return *util::GameLogicUtils::getRandomItemFromContainer(ret);
+        auto allActions = generateMovementActions(s, activeCharacter);
+        allActions.insert(allActions.end(), generateRetire(activeCharacter));
+
+        auto activeCharPosition = s.getCharacters().findByUUID(activeCharacter)->getCoordinates();
+        if (!activeCharPosition.has_value()) {
+            return {};
+        }
+        if (s.getMap().getField(activeCharPosition.value()).isFoggy()) {
```

Comment:
```suggestion
        if (!s.getMap().getField(activeCharPosition.value()).isFoggy()) {
```

Sonst generiert man gerade im Fog alle möglichen Aktionen und bewegt sich außerhalb nur noch.